modules/zivid/frame_2d.py

```diff
@@ -0,0 +1,91 @@
+"""Contains the Frame class."""
+import _zivid
+
+import zivid._settings_2d_converter as _settings_converter
+import zivid._camera_state_converter as _camera_state_converter
+import zivid._frame_info_converter as _frame_info_converter
+from zivid.image import Image
+
+
+class Frame2D:
+    """A 2D frame captured by a Zivid camera.
+
+    Contains a 2D image as well as metadata, settings and state of the API at the time of capture.
+    """
+
+    def __init__(self, internal_frame_2d):
+        """Can only be initialized by an internal Zivid 2D frame.
+
+        Args:
+            internal_frame_2d: internal 2D frame
+
+        """
+        if isinstance(internal_frame_2d, _zivid.Frame2D):
+            self.__impl = internal_frame_2d
+        else:
+            raise TypeError(
+                "Unsupported type for argument internal_frame_2d. Got {}, expected {}.".format(
+                    type(internal_frame_2d).__name__, _zivid.Frame2D
+                )
+            )
+
+    def __str__(self):
+        return str(self.__impl)
+
+    def image(self):
```

Comment:
This wrapper does not seem to have the same pixelformat template/generic parameter that the C++ and .NET wrapper has. What if we add RGB8 or RGBA16, how will that be handled in the Python wrapper?

It could be that this method shuld have a parameter to specify the pixel format, or maybe it should be named imageRGBA8().  Then we can add other image* methods when we have more formats.

Reply:
What we discussed earlier was to have only one function, but with a string parameter for datatype and then a data class where those strings can be found.

Multiple functions seems less nice.

Reply:
We can easily extend this by adding a keyword-argument for pixel-format when we need it.

Reply:
ok sounds good